istioctl/pkg/wait/wait.go

```diff
@@ -210,9 +211,17 @@ func poll(ctx cli.Context,
 		printVerbosef(cmd, "sync status: %+v", configVersions)
 		sdcnum += len(configVersions)
 		for _, configVersion := range configVersions {
-			countVersions(versionCount, configVersion.ClusterVersion)
-			countVersions(versionCount, configVersion.RouteVersion)
-			countVersions(versionCount, configVersion.ListenerVersion)
+			// TODO: support ambient ztunnel
+			if configVersion.ProxyType == model.SidecarProxy || configVersion.ProxyType == model.Router || configVersion.ProxyType == model.Waypoint {
+				countVersions(versionCount, configVersion.ClusterVersion)
+				// for gateway or sidecar, route version may be empty if no http route is configured.
+				// so we ignore route that has never been configured.
+				// There could be a race during the first push period that both lds and cds has been sent, while rds has not.
+				if configVersion.RouteVersion != "" {
+					countVersions(versionCount, configVersion.RouteVersion)
+				}
```

Comment:
Its not just a small race, this is supposed until its fully applied. If you look at the history of wait, actually a lot of it was about time between Envoy Recving the route and having it fully applied. Under load, this was taking >10s in many cases. So we are losing a ton of the value of wait here.

Reply:
Can confirm, the point of wait is for the caller to have better confidence that the config in question is in effect.  Wait is not a perfect implementation by any stretch, but our previous solution was to tell users to watch istio telemetry for p99 config propagation, and wait that long.  Wait is a marked improvement over that process.  I'd be happy to see improvements to istioctl wait, but this appears to reduce the usefulness of the command.

Can you explain what is motivating this change?  Is there a memory leak somewhere, or a bug that triggered this work?